grpc.go

```diff
@@ -364,8 +364,13 @@ func (a *agentGRPC) CreateContainer(ctx context.Context, req *pb.CreateContainer
 		mounts:      mountList,
 	}
 
-	err = container.container.Start(&builtProcess.process)
-	if err != nil {
+	// This lock has to be taken to make sure we let the libcontainer
+	// spawning process being reaped internally by libcontainer. This
+	// prevents our subreaper from doing it.
```

Comment:
IIUC the libcontainer spawning process is put on hold until libcontainer.Exec() is called. Which process does the subreaper lock protect here?

Reply:
the target process may be failed and exit or be killed. the subreaper lock is needed in this case.